esp32/modules/services.py

```diff
@@ -1,59 +1,251 @@
-import uos
-
 # File: services.py
+# Version: 2
+# API version: 1
 # Description: Background services for SHA2017 badge
 # License: MIT
 # Authors: Renze Nicolai <renze@rnplus.nl>
 #          Thomas Roos   <?>
 
-def setup():
+import uos, ujson, easywifi, easyrtc, time, appglue, deepsleep, ugfx, badge, machine
+
+services = [] #List containing all the service objects
+loopCallbacks = {} #Dict containing: {<FUNCTION>:<Wifi required on next run>}
+drawCallbacks = [] #List containing draw functions
+
+def setup(pmCb=False, drawCb=False):
     global services
+    global loopCallbacks
+    global drawCallbacks
+    
+    if pmCb:
+        print("[SERVICES] Power management callback registered")
+        global pmCallback
+        pmCallback = pmCb
```

Comment:
+ power_management is not in the scope of services.py but in the scope of splash.py so we do need to send a pointer to our function to services at some point